test/sanity/globalConstants.wtest

```diff
@@ -0,0 +1,103 @@
+const x = 1 + 1
+test "global constants are referenceable" {
+  assert.equals(2, x)
+}
+
+
+const a = b
+const b = 1 + 1
+const c = b
+test "global constants referencing other constants defined before/after are properly initialized" {
+  assert.equals(2, a)
+  assert.equals(2, b)
+  assert.equals(2, c)
+}
+
+const o = p
+object p {
+  const property e = self.f()
```

Comment:
This is specially hard... Besides cheese it because of the "self" reference, I have no idea how to detect this is a getter for a property and not any other message call

Reply:
Referencing another property is dubious. Maybe if you are a fan of double encapsulation. Personally I would replace by simple variable access

Reply:
In fact it is more interesting if it is a complex method that computes something.